lib/cli/src/commands/app/secrets/create.rs

```diff
@@ -0,0 +1,271 @@
+        if let Some(app_id) = &self.app_id.app {
+            let app = app_id.resolve(client).await?;
+            return Ok(app.id.into_inner());
+        }
+
+        let app_dir_path = if let Some(app_dir_path) = &self.app_dir_path {
+            app_dir_path.clone()
+        } else {
+            current_dir()?
+        };
+
+        if let Ok(r) = get_app_config_from_dir(&app_dir_path) {
+            let (app, _) = r;
+
+            let app_name = if let Some(owner) = &app.owner {
+                format!("{owner}/{}", app.name)
+            } else {
+                app.name.to_string()
+            };
+
+            let id = if let Some(id) = &app.app_id {
+                Some(id.clone())
+            } else if let Ok(app_ident) = AppIdent::from_str(&app_name) {
+                if let Ok(app) = app_ident.resolve(client).await {
+                    Some(app.id.into_inner())
+                } else {
+                    if !self.quiet {
+                        eprintln!("{}: the app found in {} does not exist.\n{}: maybe it was not deployed yet?", 
+                            "Warning".bold().yellow(), 
+                            format!("'{}'", app_dir_path.display()).dimmed(), 
+                            "Hint".bold());
+                    }
+                    None
+                }
+            } else {
+                None
+            };
+
+            if let Some(id) = id {
+                if !self.quiet {
+                    if let Some(owner) = &app.owner {
+                        eprintln!(
+                            "Managing secrets related to app {} ({owner}).",
+                            app.name.bold()
+                        );
+                    } else {
+                        eprintln!("Managing secrets related to app {}.", app.name.bold());
+                    }
+                }
+                return Ok(id);
+            }
+        } else if let Some(path) = &self.app_dir_path {
+            anyhow::bail!(
+                "No app configuration file found in path {}.",
+                path.display()
+            )
+        }
+
+        if self.non_interactive {
+            anyhow::bail!("No app id given. Provide one using the `--app` flag.")
+        } else {
+            let id = prompt_app_ident("Enter the name of the app")?;
+            let app = id.resolve(client).await?;
+            Ok(app.id.into_inner())
+        }
+    }
+
+    /// Given a list of secrets, checks if the given secrets already exist for the given app and
+    /// returns a list of secrets that must be upserted.
+    async fn filter_secrets(
+        &self,
+        client: &WasmerClient,
+        app_id: &str,
+        secrets: Vec<Secret>,
+    ) -> anyhow::Result<Vec<Secret>> {
+        let names = secrets.iter().map(|s| &s.name);
+        let app_secrets =
+            wasmer_api::query::get_all_app_secrets_filtered(client, app_id, names).await?;
```

Comment:
Should this kind of logic really be in the CLI?
That makes it hard to update.

This also opens the door to race conditions, so it's not actually a guarantee that create actually creates.


I feel like the backend should handle this.
Eg with a "createNew" argument to the mutation.

Reply:
This is something we thought about. The backend exposes an `upsert` mutation, abd adding a `createNew` would make it semantically inconsistent, in my opinion. (cc: @ayys)

Reply:
Created a followup issue to figure this out, we can merge without it.